File: leetcode/C++实现/0100.cpp
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q)
    {
        if(p==NULL && q==NULL)
            return true;
        else
            if(p==NULL && q!=NULL)
                return false;
        else
            if(p!=NULL  && q==NULL)
                return false;
        return (p->val==q->val && isSameTree(p->left,q->left) && isSameTree(p->right,q->right));
           
    }
};

// 一个更优化的代码 2行 其实想法一样
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q)
    {
        if(p==NULL || q==NULL)
            return p==q;
        return (p->val==q->val && isSameTree(p->left,q->left) && isSameTree(p->right,q->right));
           
    }
};

File: leetcode/JavaScript实现/100-199题/0100.js
// 自己代码
// 第一想法，递归
var isSameTree = function(p, q) {
    if(p === null && q === null)
        return true;
    if(p === null || q === null)
        return false;
    if(p.val !== q.val)
        return false;
    return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
};
